x-pack/plugins/cross_cluster_replication/public/app/components/remote_clusters_form_field.js

```diff
@@ -0,0 +1,251 @@
+              defaultMessage="View remote clusters"
+            />
+          </EuiButton>
+        </EuiCallOut>
+      );
+    };
+
+    renderRemoteClusterDoesNotExist = (name) => {
+      const { intl } = this.props;
+      const title = intl.formatMessage({
+        id: 'xpack.crossClusterReplication.forms.remoteClusterErrorTitle',
+        defaultMessage: 'Remote cluster error'
+      });
+
+      return (
+        <EuiCallOut
+          title={title}
+          color="warning"
+          iconType="cross"
+        >
+          <p>
+            { this.errorMessages.remoteClusterDoesNotExist(name) }
+          </p>
+          <EuiButton
+            {...routing.getRouterLinkProps('/', BASE_PATH_REMOTE_CLUSTERS)}
+            color="warning"
+          >
+            <FormattedMessage
+              id="xpack.crossClusterReplication.forms.viewRemoteClusterButtonLabel"
+              defaultMessage="View remote clusters"
+            />
+          </EuiButton>
+        </EuiCallOut>
+      );
+    }
+
+    renderErrorMessage = () => {
+      const { selected, remoteClusters, isEditable } = this.props;
+      const remoteCluster = remoteClusters.find(c => c.name === selected);
+      const hasClusters = Boolean(remoteClusters.length);
+      const isSelectedRemoteClusterConnected = remoteCluster && remoteCluster.isConnected;
+
+      if (isEditable) {
+        /* Create */
+        if (hasClusters && !isSelectedRemoteClusterConnected) {
+          return this.renderCurrentRemoteClusterNotConnected(selected);
+        } else if (!hasClusters) {
+          return this.renderNoClusterFound();
+        }
+      } else {
+        /* Edit */
+        const doesExists = !!remoteCluster;
+        if (!doesExists) {
+          return this.renderRemoteClusterDoesNotExist(selected);
+        } else if (!isSelectedRemoteClusterConnected) {
+          return this.renderCurrentRemoteClusterNotConnected(selected);
+        }
+      }
+
+      return null;
+    }
+
+    render() {
+      const { remoteClusters, selected, isEditable, areErrorsVisible } = this.props;
+      const remoteCluster = remoteClusters.find(c => c.name === selected);
+      const hasClusters = Boolean(remoteClusters.length);
+      const isSelectedRemoteClusterConnected = remoteCluster && remoteCluster.isConnected;
+      const isInvalid = areErrorsVisible && (!hasClusters || !isSelectedRemoteClusterConnected);
+
+      return (
+        <EuiFormRow
+          label={(
+            <FormattedMessage
+              id="xpack.crossClusterReplication.autoFollowPatternForm.remoteCluster.fieldClusterLabel"
+              defaultMessage="Remote cluster"
+            />
+          )}
+          isInvalid={isInvalid}
+          fullWidth
+        >
+          <Fragment>
+            { !isEditable && this.renderNotEditable() }
+            { isEditable && hasClusters && this.renderDropdown() }
+            <EuiSpacer size="s" />
+            { this.renderErrorMessage() }
```

Comment:
I think we can improve the UI by rendering the error as part of the dropdown. In the screenshot below I've made these changes:

1. Put the callout below the dropdown, but above the "Add" button. I think this hierarchy makes sense because the callout pertains to the selected remote cluster, but not to the button.
2. Simplified the callout copy and moved some body text into the title.
3. Changed the callout button to "Edit remote cluster". I think it makes more sense to prompt the user to edit the selected remote cluster than to view all remote clusters in this case.
4. Change the "Add" button text to "Add new remote cluster", to make it clearer that you're adding a _new_ remote cluster, and not somehow adding the one selected in the dropdown.
5. Make the button flush left to align it with the dropdown and callout, and added an icon.

![image](https://user-images.githubusercontent.com/1238659/51272964-6905b200-1980-11e9-9cca-98e6ba4c8449.png)